// react-redux-datatable/lib/shapes/TableSettingsShape.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _propTypes = require('prop-types');

var _propTypes2 = _interopRequireDefault(_propTypes);

var _airbnbPropTypes = require('airbnb-prop-types');

var _TableColumnsShape = require('./TableColumnsShape');

var _TableColumnsShape2 = _interopRequireDefault(_TableColumnsShape);

var _constants = require('../constants');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.default = {
  keyField: _propTypes2.default.string.isRequired,
  tableColumns: _TableColumnsShape2.default.isRequired,
  tableID: _propTypes2.default.string.isRequired,
  defaultSearch: _propTypes2.default.string,
  defaultSort: _propTypes2.default.arrayOf(_propTypes2.default.string, _propTypes2.default.oneOf(_constants.SORT_TYPES)),
  displayTitle: _propTypes2.default.string,
  extraToolbarItems: _propTypes2.default.func,
  extraButtons: _propTypes2.default.func,
  minWidth: _airbnbPropTypes.nonNegativeInteger,
  noDataIndication: _propTypes2.default.node,
  useLocalStorage: _propTypes2.default.bool,
  wrapperType: _propTypes2.default.string
};
